.github/workflows/scheduled.yml

```diff
@@ -71,46 +70,161 @@ env:
 jobs:
   compile:
     name: Build
-    runs-on: 8-core
+    runs-on: 16-core
+    container: ghcr.io/facebookincubator/velox-dev:centos8
     timeout-minutes: 120
     env:
-      CCACHE_DIR: "${{ github.workspace }}/.ccache/"
-      CCACHE_BASEDIR: "${{ github.workspace }}"
+      CCACHE_DIR: "/__w/velox/velox/.ccache/"
       LINUX_DISTRO: "ubuntu"
+      MAKEFLAGS: "NUM_THREADS=${{ inputs.numThreads || 16 }} MAX_HIGH_MEM_JOBS=${{ inputs.maxHighMemJobs || 8 }} MAX_LINK_JOBS=${{ inputs.maxLinkJobs || 4 }}"
+
+    defaults:
+      run:
+        shell: bash
+        working-directory: velox
+    outputs:
+      bias_presto: ${{ steps.sig-check.outputs.presto }}
+      bias_spark: ${{ steps.sig-check.outputs.spark }}
+
     steps:
 
-      - name: "Restore ccache"
+      - name: Get Function Signature Stash
+        uses: assignUser/stash/restore@v1
```

Comment:
Is this something you wrote ? :) 
(A relevant question might be whats the support model if it breaks and you are on vacation ? )

Reply:
Yes, this is the alternative for `actions/cache` I talked about. 

Well worst case is you don't have a working ccache and everything will be built from scratch but that's mostly the situation without this action so... :grimacing:

The idea is to move the repo into an ASF approved/managed collection of actions but that's currently in planning without a fixed ETA.